examples/context-propagation/rxjava/src/main/java/example/armeria/contextpropagation/rxjava/MainService.java

```diff
@@ -0,0 +1,100 @@
+package example.armeria.contextpropagation.rxjava;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+import net.javacrumbs.futureconverter.java8rx2.FutureConverter;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.util.concurrent.Uninterruptibles;
+
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.reactivex.Observable;
+import io.reactivex.Scheduler;
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+
+public class MainService implements HttpService {
+
+    private static final Splitter NUM_SPLITTER = Splitter.on(',').trimResults().omitEmptyStrings();
+
+    private final WebClient backendClient;
+
+    public MainService(WebClient backendClient) {
+        this.backendClient = requireNonNull(backendClient, "backendClient");
+    }
+
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) {
+        final Scheduler contextAwareScheduler = Schedulers.from(ctx.contextAwareExecutor());
+
+        // This logic mimics using a blocking method, which would usually be something like a MySQL
+        // database query using JDBC.
+        final Observable<Long> fetchFromFakeDb =
+                Single.fromCallable(
+                        () -> {
+                            // The context is mounted in a thread-local, meaning it is available to all
+                            // logic such as tracing.
+                            checkState(ServiceRequestContext.current() == ctx);
+                            checkState(!ctx.eventLoop().inEventLoop());
+
+                            Uninterruptibles.sleepUninterruptibly(Duration.ofMillis(50));
+                            return ImmutableList.of(23L, -23L);
+                        })
+                      // Always run blocking logic on the blocking task executor. By using
+                      // ServiceRequestContext.blockingTaskExecutor, you also ensure the context is mounted
+                      // inside the logic (e.g., your DB call will be traced!).
+                      .subscribeOn(Schedulers.from(ctx.blockingTaskExecutor()))
+                      // Make sure callbacks still stay on the context executor using observeOn.
+                      .observeOn(contextAwareScheduler)
```

Comment:
Maybe put at chaining phase?

Reply:
Can you provide your reasoning for that? I can add as a comment and update :) I figured it's better to be here since the reason we need to `observeOn` is because of the blocking subscription, so having the two close to each other seems more readable.

Reply:
I usually let storage call/remote call layer does not include subscribeOn&observeOn, I think they usually don't need to care about the `trigger` thread or ctx and we can reuse it in non armeria-context too.
So in my situation we force write all `observeOn` at service/facade layer and check when we code do code review....